### DISCUSSION TITLE
Kimai can't send email.

### DISCUSSION THREAD

**Discussion by seleondevel** [kimai/kimai#4745] [Plugins]
The Subscriber class that implements EventSubscriberInterface does not send emails.

public function onProjectCreateEvent(){
        $this->logger->info(sprintf('onProjectCreateEvent!!'));

        $email = (new Email())
            ->from('aaa@gmail.com')
            ->to('bbb@gmail.com')           
            ->subject('Time for Symfony Mailer!')
            ->text('Sending emails is fun again!')
            ->html('<p>See Twig integration for better HTML integration!</p>');
 
        try {
            $this->mailer->send($email);

            $this->logger->info(sprintf('mail send success!!'));

        } catch (TransportExceptionInterface $e) {
            // some error prevented the email sending; display an
            // error message or try to resend the message
            $this->logger->info(sprintf($e->__toString()));

        }
    }
 
Referring to https://www.kimai.org/documentation/emails.html , I set MAILER_FROM and MAILER_URL in the .env configuration file as follows, but mail send success !! is logged in the onProjectCreateEvent event, but no mail is received in mailbox.
There is no problem with the smtp setting in the google account, I encoded url to prevent the user or password from including special characters in the MAILER_URL, and I set it up, but I don't know why I'm not getting an e-mail. Is there anything else I need to check? What should I do?  The e-mail address is arbitrarily posted with aaa or bbb, but is set to the e-mail address that you actually use.
    
MAILER_FROM=aaa@gmail.com
MAILER_URL=smtps://user:password@smtp.gmail.com:465

**Comment by seleondevel** [comment]
The smtp.gmail.com:465 firewall port was blocked. And I declared dsn in the transport below, and confirmed that the mail is sent normally.

$transport = Transport::fromDsn('smtp://xxx@user:password@smtp.gmail.com:465');
$mailer = new Mailer($transport);

**Reply by kevinpapst** [reply]
Google uses `smtps://mailbox@example.com:password@smtp.gmail.com:465`
At least with a custom domain (here `example.com`) in Workspace account, never tried it with a free gmail account.

**Reply by kankadev** [reply]
Do I have to escape the app password? Google gives me something like: abca abca abca abca

Do I have to remove the spaces or set the password between Gänsefüßchen? I tried a lot of combinations. :D

**Reply by kevinpapst** [reply]
urlencode it: https://www.kimai.org/documentation/emails.html
